### PR TITLE
Feature/upload profile image

### PR DESCRIPTION
**Pull request by kaiheinrich**

### CODE REVIEW

File: frontend/src/service/ProfileService.js [hunk 1]
```diff
@@ -9,6 +9,17 @@ const header = (token) => ({
...
+export const updateProfile = (username, name, birthday, location, skills, imageName, token) =>
+    axios.put("/api/profiles/" + username, {name, birthday, location, skills, imageName}, header(token))
+        .then(response => response.data);
+
+export const uploadProfileImage = (file, token) => {
+    const formData = new FormData();
+    formData.append('image', file);
+    return axios.post('/api/profiles/image', formData, {
+        headers: {
+            'Content-Type': 'multipart/form-data',
```
Comment (jamarob) [inline frontend/src/service/ProfileService.js:21]: What do you think about this function ;)

```
const header = (token, options = {}) => ({
  headers: {
    `Bearer ${token}`,
    ...options
})
```

File: frontend/src/profiles/ProfileOverviewItem.js [hunk 2]
```diff
@@ -37,11 +39,11 @@ export default function ProfileOverviewItem({profile}) {
 const useStyles = makeStyles({
     root: {
         minWidth: 275,
-        backgroundColor: "#ffffff",
+        backgroundColor: "#FFF4F4",
         borderRadius: "20px"
     },
     button: {
-        backgroundColor: "#ec5864",
+        backgroundColor: "#9F0D2F",
```
Comment (jamarob) [inline frontend/src/profiles/ProfileOverviewItem.js:46]: Can you remember these colors? Think about defining css vars.

File: frontend/src/profiles/ProfileDetails.js [hunk 3]
```diff
@@ -25,10 +26,12 @@ export default function ProfileDetails(){
             <MenuAppBar pagename="Profile details"/>
             <ProfileDetailsStyled>
                 <div>
+                    <Card className={classes.card}>
                     <h1>{profile.name}</h1>
                     <h3>{profile.location}</h3>
                     <h4>{profile.birthday}</h4>
                     <SkillList skills={profile.skills}/>
```
Comment (jamarob) [inline frontend/src/profiles/ProfileDetails.js:33]: -semantic gap_ between `h1` and `h2`. Maybe use `address` and `time` tag instead?

File: frontend/src/profiles/EditProfile.js [hunk 4]
```diff
@@ -98,18 +107,26 @@ export default function EditProfile() {
...
+            profileData.imageName,
             token)
             .then(() => history.push("/profile/"+ profileData.username+ "/edit")).catch(error => console.log(error));
     }
 
+    function handlePictureChange(event) {
+        const imageFile = event.target.files[0];
+        imageFile ? uploadProfileImage(imageFile, token)
+            .then(data => setProfileData({...profileData, imageName: data}))
+            .catch(error => console.log(error)) : setProfileData({...profileData, imageName: profileData.imageName})
```
Comment (jamarob) [inline frontend/src/profiles/EditProfile.js:119]: Why the ternay? ´setProfileData({...profileData, imageName: profileData.imageName})´ sets the `imageName` property to the value it was before. If you just want to do it conditionally you can use `imageFile && upload...` or a simple `if`

File: backend/src/test/java/de/kaiheinrich/projectplace/service/ProjectServiceTest.java [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/service/ProjectServiceTest.java:1]: 🙀

File: backend/src/test/java/de/kaiheinrich/projectplace/service/ProfileServiceTest.java [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/service/ProfileServiceTest.java:1]: 🙀

File: backend/src/test/java/de/kaiheinrich/projectplace/security/JwtAuthFilterTest.java [hunk 7]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/security/JwtAuthFilterTest.java:1]: 🙀

File: backend/src/test/java/de/kaiheinrich/projectplace/controller/ProjectControllerIntegrationTest.java [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/controller/ProjectControllerIntegrationTest.java:1]: 😱

File: backend/src/test/java/de/kaiheinrich/projectplace/controller/ProfileControllerIntegrationTest.java [hunk 9]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/controller/ProfileControllerIntegrationTest.java:1]: 🙀

File: backend/src/test/java/de/kaiheinrich/projectplace/controller/LoginControllerTest.java [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/*
```
Comment (jamarob) [inline backend/src/test/java/de/kaiheinrich/projectplace/controller/LoginControllerTest.java:1]: 🙀

File: backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java [hunk 11]
```diff
@@ -12,15 +20,48 @@
...
+        List<Profile> profileList = profileDb.findAll();
+
+        java.util.Date expiration = new java.util.Date();
+        long expTimeMillis = expiration.getTime();
+        expTimeMillis += 1000 * 60 * 60;
+        expiration.setTime(expTimeMillis);
+
+        for(Profile profile : profileList) {
+
+            if(!profile.getImageName().equals("")){
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java:56]:
```suggestion
            if(profile.getImageName().isBlank()){
```
Looks much nicer , imho.

File: backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java [hunk 12]
```diff
@@ -12,15 +20,48 @@
...
     public ProfileService(ProfileMongoDb profileDb) {
         this.profileDb = profileDb;
     }
 
     public List<Profile> getProfiles() {
-        return profileDb.findAll();
+
+        Regions clientRegion = Regions.EU_CENTRAL_1;
+        BasicAWSCredentials awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
+        String bucketName = "kais-super-bucket";
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java:41]: That's the best bucket :D Maybe put the bucket name into an environment variable to make it configurable without recompiling?

File: backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java [hunk 13]
```diff
@@ -12,15 +20,48 @@
...
+
+        AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
+                .withRegion(clientRegion).withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+                .build();
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java:45]: I think you can put these outside of the method 🤔 they don't have to be redefined on every call

File: backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java [hunk 14]
```diff
@@ -12,15 +20,48 @@
...
+
+        List<Profile> profileList = profileDb.findAll();
+
+        java.util.Date expiration = new java.util.Date();
+        long expTimeMillis = expiration.getTime();
+        expTimeMillis += 1000 * 60 * 60;
+        expiration.setTime(expTimeMillis);
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileService.java:52]: Maybe put this in a nice small helper method `getExpirationTIme()`?

File: backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileImageUploadAWSService.java [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+
+import java.io.IOException;
+import java.net.URL;
+
+@Service
+public class ProfileImageUploadAWSService {
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/service/ProfileImageUploadAWSService.java:22]: I think this is geeraic enough to be simply names `ImageUploadService`

File: backend/src/main/java/de/kaiheinrich/projectplace/security/SecurityConfig.java [hunk 16]
```diff
@@ -37,6 +37,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 .authorizeRequests()
                 .antMatchers("/api/**").authenticated()
                 .antMatchers("/**").permitAll()
+                //.antMatchers("/api/profiles/image").permitAll()
```
Comment (jamarob) [inline backend/src/main/java/de/kaiheinrich/projectplace/security/SecurityConfig.java:40]: Please delete comments from main

**Review by jamarob — COMMENTED** [review]
Yay images 🎉 

Left a a few comments for improving code quality and I think you forgot to delete a lot of comments from your code 😉

File: frontend/src/profiles/EditProfile.js [hunk 17]
```diff
@@ -43,30 +46,35 @@ export default function EditProfile() {
             <MenuAppBar pagename="Edit profile"/>
             <MainStyled>
                 <FormStyled onSubmit={handleSubmit}>
+                    {profileData.imageName ? <Avatar alt="profile" src={profileData.imageUrl}/> : <Avatar/>}
+                    <input type="file" onChange={handlePictureChange} />
```
Comment (christophersiem) [inline frontend/src/profiles/EditProfile.js:50]: with `accept="image/*"` only images can be uploaded

```suggestion
                    <input type="file" onChange={handlePictureChange} accept="image/*" />
```

**Pull request merged**